### PR TITLE
a single pymarl_MA file

### PR DESCRIPTION
**Pull request by PaLeroy**

### CODE REVIEW

**Comment by PaLeroy** [comment]
still some changes to do before removing it :)

**Comment by moratodpg** [comment]
This is great. Could we additionally make this customed parameter more independent?
I will add another environment this week.
In that one, we have to specify other specific config parameters (perhaps more than one).

**Comment by PaLeroy** [comment]
Yeah.
In fact, I think it would be better if we just use a `dict` of additional parameters.

**Comment by PaLeroy** [comment]
How is this new version?
Before merging, we'll have to change config files and so on to suite with the custom_param dictionary.

**Comment by PaLeroy** [comment]
I also think that we need to create an interface (like [multi agent envs](main/pymarl/envs/multiagentenv.py)) for a struct env.
`struct_env` and `owf` would inherit from this interface.
What do you think?
So that people can easily integrate their structural problem (like our struct_env and owf_env) without going into the details of [pymarl_ma_struct](main/struct_env/pymarl_ma_struct.py) and [pymarl_sa_struct](main/struct_env/pymarl_sa_struct.py).

**Comment by moratodpg** [comment]
I like this implementation very much! Also, creating an interface seems a reasonable and nice idea.
Let's then (i) change config files and (ii) in the near future, create an interface. :)

File: README.md [hunk 1]
```diff
@@ -21,5 +21,5 @@ and
 ## Run an experiment 
 
 ```shell
-python3 main.py --config=qmix --env-config=struct with env_args.n_comp=10 env_args.k_comp=9
+python3 main.py --config=qmix --env-config=struct with env_args.n_comp=10 env_args.custom_param=9
```
Comment (moratodpg) [inline README.md]: This will now change?

**Comment by PaLeroy** [comment]
Agreed that the interface will be the next thing after the paper :D

**Comment by PaLeroy** [comment]
Check I did not miss something but I guess we're fine :)

**Review by moratodpg — APPROVED** [review]

**Comment by moratodpg** [comment]
I have not noticed anything to be further reviewed, shall we also merge this one?

File: README.md (hunk 1 above)
Comment (PaLeroy) [reply README.md]: Yep nice catch!

**Comment by PaLeroy** [comment]
Ok to merge for me.

**Pull request merged**